### PR TITLE
Windows: Restore some old settings

### PR DESCRIPTION
**Pull request by zerok**
During testing of the new nsis-based installer I discovered a couple of differences from the old version and tried to restore the old behaviour:

- Uninstalling should also remove the app from the "Add and remove..." listing again
- The service name was originally "Grafana" and has now been restored to it again
- Restored the old AppNames ("GrafanaOSS", "GrafanaEnterprise") to keep testing and docs consistent
- Use the icon also for the uninstaller

Additionally, I've added a check that prevent GrafanaEnterprise from being installed over GrafanaOSS and vice versa.

### CODE REVIEW

**Comment by kminehart** [comment]
god I hate the naming "GrafanaOSS". It's just Grafana.

i guess we have to leave it like this for now. It would make me so happy to get away from this naming though.

**Review by kminehart — CHANGES_REQUESTED** [review]

File: pipelines/package_names.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ func TarFilename(opts TarFileOpts) string {
 }
 
 func TarOptsFromFileName(filename string) TarFileOpts {
+	filename = filepath.Base(filename)
```
Comment (kminehart) [inline pipelines/package_names.go:57]: you may want to merge with main on this, this function had some improvements to handle non-tar.gz extensions and might affect your changes.

Comment (kminehart) [reply pipelines/package_names.go:57]: oh nvm I see you did already oops

File: scripts/packaging/windows/grafana.nsis [hunk 2]
```diff
@@ -9,6 +9,13 @@ InstallDir $PROGRAMFILES64\GrafanaLabs\Grafana
 
 Outfile "grafana-setup.exe"
 Name "${APPNAME}"
+!if ${APPNAME} == "GrafanaOSS"
+    !define CONFLICT_APP "GrafanaEnterprise"
+    !define CONFLICT_APP_NAME "Grafana Enterprise"
+!else
+    !define CONFLICT_APP "GrafanaOSS"
+    !define CONFLICT_APP_NAME "Grafana OSS"
+!endif
```
Comment (kminehart) [inline scripts/packaging/windows/grafana.nsis:18]: hahahaha you didn't have to do this but this does make me happy.

**Review by kminehart — APPROVED** [review]
Love it!

**Comment by kminehart** [comment]
I'm going to merge this for you since @guicaulada opened a conflicting PR. I'd rather that he resolve those conflicts in his PR than make you wait even longer to get this merged.

**Comment by zerok** [comment]
Thank you :) Forgot to set automerge -_-

**Pull request merged**